src/CompletionProvider.ts

```diff
@@ -11,20 +11,136 @@ import {
 } from './utils';
 import type {CamelCaseValues} from './utils';
 
-// check if current character or last character is .
-function isTrigger(line: string, position: Position): boolean {
+export const COMPLETION_TRIGGERS = ['.', '[', '"', "'"] as const;
+
+type FieldOptions = {
+    wrappingBracket: boolean;
+    startsWithQuote: boolean;
+    endsWithQuote: boolean;
+};
+
+/**
+ * check if current character or last character is any of the completion triggers (i.e. `.`, `[`) and return it
+ *
+ * @see COMPLETION_TRIGGERS
+ */
+function findTrigger(line: string, position: Position): string | undefined {
```

Comment:
nit: void should be preferred in return types if we do not intend to use it as a value
```suggestion
function findTrigger(line: string, position: Position): string | void {
```

Reply:
I do get the following lint error when applying this suggestion:
```
void is not valid as a constituent in a union type [lint/suspicious/noConfusingVoidType]
```
I think I could try and disable that rule, but wanted to double check your thoughts here.

Reply:
oh that's really interesting. I was not aware of this lint rule. Let's leave it as `undefined` then. From biome [docs](https://biomejs.dev/linter/rules/no-confusing-void-type/). it looks like they allow void as a standalone return type but not as union in return. Thanks for checking it out.